ExclusionFile for LinkageCheckerRule needs to have correct setter name.

To suppress linkage errors using [Linkage Checker Exclusion File](https://github.com/GoogleCloudPlatform/cloud-opensource-java/wiki/Linkage-Checker-Exclusion-File), set `exclusionFile` element to the exclusion file.

```xml
  <LinkageCheckerRule
      implementation="com.google.cloud.tools.dependencies.enforcer.LinkageCheckerRule">
      <exclusionFilterFile>linkage-checker-exclusion-rules.xml</exclusionFilterFile>
  </LinkageCheckerRule>
```

It gives the following error:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M3:enforce
 (enforce-linkage-checker) on project no-such-method-error-example: Unable to parse configuration
 of mojo org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M3:enforce for parameter 
exclusionFilterFile: Cannot find 'exclusionFilterFile' in class 
com.google.cloud.tools.dependencies.enforcer.LinkageCheckerRule -> [Help 1]
```

Setting `<exclusionFile>` element also fails:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M3:enforce 
(enforce-linkage-checker) on project no-such-method-error-example: Unable to parse configuration
 of mojo org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M3:enforce for parameter exclusionFile:
 Cannot create instance of interface java.nio.file.Path: java.nio.file.Path.<init>() -> [Help 1]
[ERROR] 

```

--- a/enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java
+++ b/enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java
@@ -103,7 +103,7 @@ public class LinkageCheckerRule extends AbstractNonCacheableEnforcerRule {
    */
   private boolean reportOnlyReachable = false;
 
-  private Path exclusionFile = null;
+  private String exclusionFile = null;
 
   private ClassPathBuilder classPathBuilder = new ClassPathBuilder();
 
@@ -126,8 +126,8 @@ public class LinkageCheckerRule extends AbstractNonCacheableEnforcerRule {
   }
 
   @VisibleForTesting
-  void setExclusionFilterFile(String exclusionFileLocation) {
-    exclusionFile = Paths.get(exclusionFileLocation);
+  void setExclusionFile(String exclusionFile) {
+    this.exclusionFile = exclusionFile;
   }
 
   private Log logger;
@@ -195,6 +195,8 @@ public class LinkageCheckerRule extends AbstractNonCacheableEnforcerRule {
         // TODO LinkageChecker.create and LinkageChecker.findSymbolProblems
         // should not be two separate public methods since we all call
         // findSymbolProblems immediately after create
+
+        Path exclusionFile = this.exclusionFile == null ? null : Paths.get(this.exclusionFile);
         LinkageChecker linkageChecker =
             LinkageChecker.create(classpath, entryPoints, exclusionFile);
         ImmutableSetMultimap<SymbolProblem, ClassFile> symbolProblems =

--- a/enforcer-rules/src/test/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRuleTest.java
+++ b/enforcer-rules/src/test/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRuleTest.java
@@ -442,7 +442,7 @@ public class LinkageCheckerRuleTest {
           Paths.get(ClassLoader.getSystemResource("appengine-exclusion.xml").toURI())
               .toAbsolutePath()
               .toString();
-      rule.setExclusionFilterFile(exclusionFileLocation);
+      rule.setExclusionFile(exclusionFileLocation);
       rule.execute(mockRuleHelper);
       Assert.fail(
           "The rule should raise an EnforcerRuleException for artifacts missing dependencies");

--- a/example-problems/no-such-method-error-signature-mismatch/pom.xml
+++ b/example-problems/no-such-method-error-signature-mismatch/pom.xml
@@ -47,7 +47,7 @@
           <dependency>
             <groupId>com.google.cloud.tools</groupId>
             <artifactId>linkage-checker-enforcer-rules</artifactId>
-            <version>1.1.5-SNAPSHOT</version>
+            <version>1.1.4</version>
           </dependency>
         </dependencies>
         <executions>

--- a/example-problems/no-such-method-error-signature-mismatch/pom.xml
+++ b/example-problems/no-such-method-error-signature-mismatch/pom.xml
@@ -47,7 +47,7 @@
           <dependency>
             <groupId>com.google.cloud.tools</groupId>
             <artifactId>linkage-checker-enforcer-rules</artifactId>
-            <version>1.1.1</version>
+            <version>1.1.5-SNAPSHOT</version>
           </dependency>
         </dependencies>
         <executions>

